modin/distributed/dataframe/pandas/partitions.py

```diff
@@ -159,8 +176,24 @@ def from_partitions(partitions, axis):
             f"Got unacceptable value of axis {axis}. Possible values are {0}, {1} or {None}."
         )
 
-    index = partition_mgr_class.get_indices(0, parts, lambda df: df.axes[0])
-    columns = partition_mgr_class.get_indices(1, parts, lambda df: df.axes[1])
-    return DataFrame(
-        query_compiler=PandasQueryCompiler(partition_frame_class(parts, index, columns))
+    sync_labels_on = None
+    if index is None:
+        sync_labels_on = 1
+        index = partition_mgr_class.get_indices(0, parts, lambda df: df.axes[0])
+
+    if columns is None:
+        sync_labels_on = 0 if sync_labels_on is None else -1
+        columns = partition_mgr_class.get_indices(1, parts, lambda df: df.axes[1])
```

Comment:
```suggestion
    new_axes = [
        partition_mgr_class.get_indices(i, parts, lambda df: df.axes[i])
        if axis is None
        else axis
        for i, axis in enumerate([index, columns])
    ]
```

Reply:
What's need of propagating `index`/`columns` received from partitions after `partition_mgr_class.get_indices`? Partitions already have the correct `index`/`columns` in that case.